x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/pki/PkiRealm.java

```diff
@@ -149,9 +149,8 @@ public void authenticate(AuthenticationToken authToken, ActionListener<Authentic
     }
 
     private void buildUser(X509AuthenticationToken token, ActionListener<AuthenticationResult> listener) {
-        final Map<String, Object> metadata = Collections.singletonMap("pki_dn", token.dn());
-        final UserRoleMapper.UserData userData = new UserRoleMapper.UserData(token.principal(),
-                token.dn(), Collections.emptySet(), metadata, this.config);
+        final Map<String, Object> metadata = Map.of("pki_dn", token.dn());
```

Comment:
Similar to my comments on empty collections, I think we need to reach a team-wide decision on `singletonXXX` vs `of`

Reply:
I think it would be inconsistent to use `Collections.singletonMap` for a single entry but `Map.of` for several. Similarly, it would be inconsistent to use `Collections.emptyMap()` . So if we should use `Map.of` for two entries, then we should use it for one or the empty map, right?